### PR TITLE
Add Cause to API.

### LINKED ISSUE #23: CloudManagerApiException message contains double slash in URL
The messages exposed by `CloudManagerApiException` emit the url 

```io.adobe.cloudmanager.CloudManagerApiException: Cannot get execution: https://cloudmanager.adobe.io//api/program/.../pipeline/.../execution (404 Not Found)```

### Expected Behaviour

There should be no double slash in the URL after the host name.

### Actual Behaviour

There is a double slash due to the fact that baseUrl ends with a slash and path starts with a slash. 

Although the actual request works fine, a correct URL should be used both for the request as well as for the exception.

### PR DESCRIPTION
**Pull request by bstopp**
## Description

Provide the root API cause to consumers. 

## Related Issue
Resolves #30 
Fixes #23 

## How Has This Been Tested?

Run existing test cases.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/main/java/io/adobe/cloudmanager/CloudManagerApiException.java [hunk 1]
```diff
@@ -85,6 +88,12 @@ public CloudManagerApiException(ErrorType type, String baseUrl, String apiPath,
    */
   public CloudManagerApiException(ErrorType type, String... vars) {
     this.message = String.format(type.message, (Object[]) vars);
+    this.cause = null;
+  }
+
+  @Override
+  public ApiException getCause() {
```
Comment (kwin) [inline src/main/java/io/adobe/cloudmanager/CloudManagerApiException.java:95]: This makes the API even more dependent on Swagger internal classes. That makes switching to another client generator much harder in the future. I would recommend to expose as less as possible Swagger specific classes.

Comment (bstopp) [reply src/main/java/io/adobe/cloudmanager/CloudManagerApiException.java:95]: The _entire_ solution is based on the swagger generated classes. Every API response is either a Swagger class, or a wrapper around a swagger class.

Comment (kwin) [reply src/main/java/io/adobe/cloudmanager/CloudManagerApiException.java:95]: There is IMHO a difference between model classes (which are mainly based on the spec which is stable) and the ApiException (which is a Java-Client specific implementation detail of the Swagger generator)

**Comment by kwin** [comment]
@bstopp Can we somehow continue here (either by merging this or #30)?

The description refers to the wrong issue though. This fixes #16 but not #23.

**Pull request merged**